### PR TITLE
add function and tests

### PR DESCRIPTION
**Pull request by jordanlau123**

### CODE REVIEW

File: tests/test_trie_add.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+def test_add():
+    trie = Trie()
+    assert ((trie.add("hello")) == True)
+
+def test_add_existing_word():
+    trie = Trie()
+    trie.add("hello")
+    assert ((trie.add("hello")) == False)
+
+def test_word_is_string():
```
Comment (Andrew-Tan) [inline tests/test_trie_add.py]:
```suggestion
def test_add_word_is_string():
```
Let's add consistent prefixes to all tests function. For all tests related to `add()`, we can add the prefix `test_add`

File: tests/test_trie_add.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+     trie = Trie()  
+     with pytest.raises(TypeError):
+         trie.add(23)
+
+def test_empty_string():
```
Comment (Andrew-Tan) [inline tests/test_trie_add.py]:
```suggestion
def test_add_empty_string():
```

File: tests/test_trie_add.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+     trie = Trie()       
+     with pytest.raises(ValueError):
+         trie.add("")
+
+def test_sucessfully_add_word():
```
Comment (Andrew-Tan) [inline tests/test_trie_add.py]:
```suggestion
def test_add_sucessfully_add_word():
```

**Review by Andrew-Tan — CHANGES_REQUESTED** [review]
Great work! Just some minor details regarding giving consistent prefixes to all test function names.

**Comment by jordanlau123** [comment]
Thanks for editing the test names!

**Review by Andrew-Tan — APPROVED** [review]
LGTM

**Pull request merged**